src/main/config.cpp

```diff
@@ -294,12 +294,25 @@ IndexTypeSet &DBConfig::GetIndexTypes() {
 }
 
 void DBConfig::SetDefaultMaxMemory() {
-	auto memory = FileSystem::GetAvailableMemory();
-	if (!memory.IsValid()) {
-		options.maximum_memory = DBConfigOptions().maximum_memory;
+	const char *slurm_mem_per_node = getenv("SLURM_MEM_PER_NODE");
```

Comment:
Can we move this into a separate function (`DBConfig::GetSystemAvailableMemory()`) similar to `GetSystemMaxThreads`?